authorization/v2/uniformResourceIdentifier.go

```diff
@@ -157,3 +158,57 @@ func marshallURIMap(uriMap map[string]string) string {
 	// Remove the extra '&' character introduced when marshaling the uriMap
 	return marshalledMap[:len(marshalledMap)-1]
 }
+
+// isURIMatch checks that the target URI is a subset of the source URI (the URI which the user can access)
+func (uri UniformResourceIdentifier) isSubsetOf(target UniformResourceIdentifier) bool {
+	// Ensure the target path is a subset of the source path
+	if len(uri.path) > len(target.path) {
+		return false
+	}
+
+	// Compare URI path
+	for i, sourceCharacter := range uri.path {
```

Comment:
Any reason why we can't just have:
```
if uri.path != target.path {
  return false
}
```
?

Reply:
Nvm, I'm dum.

But you should be able to replace it with:
```
if strings.HasPrefix(target.path, uri.path) {
  return false
}
```
https://golang.org/pkg/strings/#HasPrefix

Reply:
Also I just realized, that this is not how it's supposed to work.

`hs:hs_auth:api:v2:GetUser` is not on the same path as `hs:hs_auth:api:v2:GetUsers`. We can't just check if one URI path is a prefix of another path.

Could you add a test case for this as well after reworking the logic?